### ISSUE TITLE
Latest version, if different from the one listed, creates double entry

### ISSUE THREAD

**Issue by ankush981** [zachtylr21/pyp#6]
I noticed that the source code comes with `pytest==6.0.2` listed in `requirements.txt`. However, if I do install `pytest` again, the latest version is different and we end up having two entries in the `requirements.txt`. It feels like we should be doing more than comparing the entire string mentioning a dependency, but I don't know the stuff very well so am not 100% sure. 😐

**Comment by zachtylr21** [comment]
That's a really good catch. We definitely should definitely be taking version number into account when we write the requirements file. Did you want to take this on? If not, no worries, I'll look into it.

**Comment by ankush981** [comment]
No, I want to do this. Or at least give it a good try! 😅 Let me think about this and come back with questions and/or an approach.

**Comment by zachdtaylor** [comment]
Did we ever fix this, or is it still an issue?

**Comment by ankush981** [comment]
Hey @zachtylr21, sorry I couldn't do anything at all these past few weeks. Life hasn't been great for me, is all I can say in brief. 🙂 Please feel free to assign this to someone else; if I can get around to finishing this, I'll let you know (maybe I'll leave a comment on this very thread?). Thanks and apologies again.

**Comment by zachdtaylor** [comment]
@ankush981 No worries! Sorry to hear things haven't been going so well, hope things start looking up soon.
